app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/HorizontalSnapScrollState.kt

```diff
@@ -53,9 +53,11 @@ data class HorizontalSnapScrollState @JvmOverloads constructor(
 
     /**
      * Returns the column index calculated based on the given [scrollX] value
-     * and the current state of [HorizontalSnapScrollState].
+     * and the current state of [HorizontalSnapScrollState]. Normaly, the output
+     * Value is clamped to the allowed columnIndex range, which can be turned of
+     * using the [clamped] parameter
      */
-    fun calculateOnTouchColumnIndex(scrollX: Int): Int {
+    fun calculateOnTouchColumnIndex(scrollX: Int, clamped:Boolean = true): Int {
```

Comment:
Why would we add the boolean flag as a public property if it is `false` in every use-case? This should be the only behavior. It looks like as if we can replace the min/max code here with the "limiting" code which is currently in `HorizontalSnapScrollView.OnTouchListener#onTouch`.

Reply:
The the min/max Code was added with the Improvements made after transpiling the class from Java to Kotlin. I thought the lack of any range limitation was just a bug, as the function name does not imply that the output might not be a valid Index.
I thought it would be a good Idea to make the default behavior of the Method as safe/intuitive as possible, to avoid bugs produced in the future.

Reply:
I would not leave code/behavior behind (the suggested default behavior, `clamped: Boolean = true`) which is **not used** anywhere (at runtime). The purpose of the function is to calculate an index which is safe to be scrolled to. The range limitation can be calculated by the current state of properties in `HorizontalSnapScrollState` as you show in your pull request. Therefore, it is an **implementation detail** to the overall calculation and should be **hidden within** `calculateOnTouchColumnIndex()`. I would not split the index calculation into two methods - the caller would _need to know_ to invoke both which is considered bad interface design.

Reply:
Updated the code, now the `calculateOnTouchColumnIndex()` returns -1 if the result would be invalid and the `OnTouch()` Method just checks if the returned Value is `>= 0`

Reply:
OK. I like this better.